text/0000-fabric-private-chaincode-1.0.md

```diff
@@ -0,0 +1,720 @@
+Extended support for other Fabric Client SDK, such as the  NodeSDK, will be future work.
+
+An application can interact with the asset store chaincode from our [Hello World Tutorial](https://github.com/hyperledger-labs/fabric-private-chaincode/tree/master/examples) using the FPC Client SDK based on the gateway API of the Fabric Client Go SDK.
+Here an example ***app.go***:
+```go
+// Get FPC Contract
+contract := fpc.GetContract(network, "helloWorld")
+
+result, err = contract.SubmitTransaction("storeAsset", "myDiamond", "100000")
+if err != nil {
+    log.Fatalf("Failed to Submit transaction: %v", err)
+}
+```
+
+The FPC Client SDK API is documented [here](https://github.com/hyperledger-labs/fabric-private-chaincode/blob/master/client_sdk/go/fpc/contract.go) in detail.
+
+## FPC Chaincode Deployment
+
+### Peer Setup 
+There are two preparation steps required before one can deploy FPC Chaincode:
+
+- The chaincode administrator has to register with the [*Intel Attestation Service (IAS)*](https://software.intel.com/content/www/us/en/develop/topics/software-guard-extensions/attestation-services.html) to obtain attestation credentials and configure the local platform correspondingly.
+- The peer operator has to add FPC in the `externalBuilders` section in `core.yaml` and restart the peer.
+  <!-- See an example `core.yaml` [here](https://github.com/hyperledger-labs/fabric-private-chaincode/blob/master/integration/config/core.yaml#L569). -->
+  FPC provides external builder and launcher scripts to run a FPC Chaincode on a peer or as an external service.
+  ```yaml
+  chaincode:
+    externalBuilders:
+      - path: /opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode/fabric/externalBuilder/chaincode_server
+        name: fpc-external-launcher
+        propagateEnvironment:
+          - CORE_PEER_ID
+  ```
+
+
+### Deployment
+The chaincode deployment follows mostly the standard Fabric procedure:
+
+- The deployer compiles an agreed-upon FPC Chaincode using the FPC build environment.
+  This step creates a package with the main FPC deployment artifact, the `enclave.signed.so` enclave binary, and deployment metadata.
+  FPC provides convenience scripts to facilitate this step.
+- Subsequently, the deployer follows the standard Fabric 2.0 Lifecycle steps, i.e., `install`, `approveformyorg` and `commit`. 
+  A noteworthy FPC specific aspect is that the version used *must* be the code identity (i.e., `MRENCLAVE`) of the FPC Chaincode.
+  ```bash
+  peer lifecycle chaincode approveformyorg --name ${CC_ID} --version ${MRENCLAVE} ...
+  peer lifecycle chaincode commit --name ${CC_ID} --version ${MRENCLAVE} ...
+  ```
+- Lastly, once the chaincode definition for an FPC Chaincode has been approved by the consortium, 
+  a Chaincode Enclave is created via a new `initEnclave` lifecycle management command. 
```

Comment:
initEnclave is a new lifecycle command in Fabric itself?

Reply:
No, the initEnclave command belongs to FPC. Fabric does not require any change to support it. In fact, under the hood, initEnclave is implemented as function of a FPC chaincode package (Go part).

Additionally, we provide the initEnclave command as part of the FPC Admin Client SDK in Go.

Reply:
ok, could you clarify this in the RFC.

Reply:
@denyeart we clarified this in the RFC.